src/brewHandler.h

```diff
@@ -72,89 +69,75 @@ boolean brewPIDDisabled = false;                    // is PID disabled for delay
  * @brief Switch or trigger input for BREW SWITCH
  */
 void checkbrewswitch() {
-    #if BREWSWITCH_TYPE == 1
-            // Digital 
-            brewSwitch = digitalRead(PIN_BREWSWITCH);
-    #endif
-
-    #if BREWSWITCH_TYPE == 2
-            int reading = digitalRead(PIN_BREWSWITCH);
-
-            if (reading != lastStateBrewTrigger) {
-                // restart the debouncing timer
-                lastDebounceTimeBrewTrigger = millis();
-                // set new button state 
-            }
-            else if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
-                // whatever the reading is at, it's been there for longer than the debounce
-                // delay, so take it as the actual current state:
-                
-                if (brewSwitchTrigger != reading) {
-                    brewSwitchTrigger = reading;
-                }
-            }
-            lastStateBrewTrigger = reading;
+    uint8_t brewSwitchReading = brewSwitch->isPressed();
 
+    if (BREWSWITCH_TYPE == Switch::TOGGLE) {
+        currStateBrewSwitch = brewSwitchReading;
+    }
+    else if (BREWSWITCH_TYPE == Switch::MOMENTARY) {
+        if (currBrewSwitchStateMomentary != brewSwitchReading) {
+            currBrewSwitchStateMomentary = brewSwitchReading;
+        }
+        
         // Convert trigger signal to brew switch state
-        switch (brewSwitchTriggerCase) {
+        switch (brewSwitchCase) {
```

Comment:
This variable might be better called something like `brewSwitchState` as this is a state machine and the variable holds the state. This might get confusing with the switch state variables however, so maybe these should not include state but rather value?

Reply:
Renamed to brewSwitchState

Reply:
See above